server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -840,7 +838,11 @@ public void run() {
                                 scheduleNow(() -> testClusterNodes.stopNode(masterNode));
                             }
                             testClusterNodes.randomDataNodeSafe().client.admin().cluster().prepareCreateSnapshot(repoName, snapshotName)
-                                .execute(snapshotStartedListener);
+                                    .execute(ActionListener.wrap(() -> {
```

Comment:
This is basically a revert of https://github.com/elastic/elasticsearch/commit/93c6d771ee3553394aafead44911a73571a754c9

Master fail-over would not lead to a failing snapshot response here with any observable frequency when we had the INIT state.
If the client (connected to a data node) lost the connection to the master node (because we disconnect the master temporarily in some runs), it would retry and the master meanwhile would have at the most made it to an `INIT` state in the CS (which the retry would simply ignore once a new master is elected). Now the first CS update will always be a `STARTED` state snapshot and a retry can again run into an in-progress snapshot. 
We should fix clean retrying some other way (could do what we did for force merge UUIDs and generate the unique `SnapshotId` in the transport request already so that retries can know they're a retry). Since we haven't released `7.8` and this is a really minor UX win to revert here, I think this is an acceptable "regression".